dataconnect/dataconnect/movie-connector/queries.gql

```diff
@@ -0,0 +1,383 @@
+    rating
+    tags
+  }
+}
+
+# Get movie by id
+query GetMovieById($id: UUID!) @auth(level: PUBLIC) {
+  movie(id: $id) {
+    id
+    title
+    imageUrl
+    releaseYear
+    genre
+    rating
+    description
+    tags
+    metadata: movieMetadatas_on_movie {
+      director
+    }
+    mainActors: actors_via_MovieActor(where: { role: { eq: "main" } }) {
+      id
+      name
+      imageUrl
+    }
+    supportingActors: actors_via_MovieActor(
+      where: { role: { eq: "supporting" } }
+    ) {
+      id
+      name
+      imageUrl
+    }
+    reviews: reviews_on_movie {
+      id
+      reviewText
+      reviewDate
+      rating
+      user {
+        id
+        username
+      }
+    }
+  }
+}
+
+# Get actor by id
+query GetActorById($id: UUID!) @auth(level: PUBLIC) {
+  actor(id: $id) {
+    id
+    name
+    imageUrl
+    mainActors: movies_via_MovieActor(where: { role: { eq: "main" } }) {
+      id
+      title
+      genre
+      tags
+      imageUrl
+    }
+    supportingActors: movies_via_MovieActor(
+      where: { role: { eq: "supporting" } }
+    ) {
+      id
+      title
+      genre
+      tags
+      imageUrl
+    }
+  }
+}
+
+# List movies by partial title match
+query ListMoviesByPartialTitle($input: String!) @auth(level: PUBLIC) {
+  movies(where: { title: { contains: $input } }) {
+    id
+    title
+    genre
+    rating
+    imageUrl
+  }
+}
+
+# List movies by tag
+query ListMoviesByTag($tag: String!) @auth(level: PUBLIC) {
+  movies(where: { tags: { includes: $tag } }) {
+    id
+    title
+    imageUrl
+    genre
+    rating
+  }
+}
+
+# List movies by release year range
+query MoviesByReleaseYear($min: Int, $max: Int) @auth(level: PUBLIC) {
+  movies(
+    where: { releaseYear: { le: $max, ge: $min } }
+    orderBy: { releaseYear: ASC }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# List movies by rating and genre with OR filters
+query SearchMovieOr(
+  $minRating: Float
+  $maxRating: Float
+  $genre: String
+  $tag: String
+  $input: String
+) @auth(level: PUBLIC) {
+  movies(
+    where: {
+      _or: [
+        { rating: { ge: $minRating } }
+        { rating: { le: $maxRating } }
+        { genre: { eq: $genre } }
+        { tags: { includes: $tag } }
+        { title: { contains: $input } }
+      ]
+    }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# List movies by rating and genre with AND filters
+query SearchMovieAnd(
+  $minRating: Float
+  $maxRating: Float
+  $genre: String
+  $tag: String
+  $input: String
+) @auth(level: PUBLIC) {
+  movies(
+    where: {
+      _and: [
+        { rating: { ge: $minRating } }
+        { rating: { le: $maxRating } }
+        { genre: { eq: $genre } }
+        { tags: { includes: $tag } }
+        { title: { contains: $input } }
+      ]
+    }
+  ) {
+    id
+    rating
+    title
+    imageUrl
+  }
+}
+
+# Get favorite movies by user ID
+query GetUserFavoriteMovies @auth(level: USER) {
+  user(key: {id_expr: "auth.uid"}) {
```

Comment:
nit: `user(id_expr: "auth.uid")` would work as well, a bit shorter

Same below.

`user(id)` and `user(id_expr)` are available because this table has a singular primary key named `id`. It's a convenient short-hand for common schema.

If the pk field is named other than `id`, they would have to use `key: {pkName}` instead.

Reply:
Gotcha! We'll put that in the documentation, and remove it here